components/TextFields/src/MDCTextInputUnderlineView.m

```diff
@@ -0,0 +1,179 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTextInputUnderlineView.h"
+
+#import "MaterialPalettes.h"
+
+NSString *const MDCTextInputUnderlineColorKey = @"MDCTextInputUnderlineColorKey";
+NSString *const MDCTextInputUnderlineEnabledKey = @"MDCTextInputUnderlineEnabledKey";
+NSString *const MDCTextInputUnderlineLineHeightKey = @"MDCTextInputUnderlineLineHeightKey";
+
+static const CGFloat MDCTextInputUnderlineDefaultHeight = 1.f;
+
+// TODO: (larche): Make disabled color parameterized?
+static inline UIColor *MDCTextInputUnderlineColor() {
+  return [UIColor lightGrayColor];
+}
+
+@implementation MDCTextInputUnderlineView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    _color = MDCTextInputUnderlineColor();
+    _enabled = YES;
+    _lineHeight = MDCTextInputUnderlineDefaultHeight;
+
+    [self commonMDCUnderlineViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+  self = [super initWithCoder:coder];
+  if (self) {
+    _color = [coder decodeObjectForKey:MDCTextInputUnderlineColorKey];
+    _enabled = [coder decodeBoolForKey:MDCTextInputUnderlineEnabledKey];
+    _lineHeight = (CGFloat)[coder decodeFloatForKey:MDCTextInputUnderlineLineHeightKey];
+
+    [self commonMDCUnderlineViewInit];
```

Comment:
Consider:
Move commonMDCInit to the beginning of the if clause so you can overwrite the values taken from the decoder.
_enabled should likely default to YES.

if ([coder containsValueForKey:MDCInputUnderlineEnabledKey]) {
  _enabled = [coder decodeBoolForKey:MDCTextInputUnderlineEnabledKey];
}